### PR TITLE
Adding flow trainer flags

### PR DESCRIPTION
**Pull request by vratins**
Adding flags during training:

- seeds
- eval method + number of steps
- selection metric (f1 vs auc vs blend)
- lr decay parameters
- resume logic

### CODE REVIEW

File: scripts/train.py [hunk 1]
```diff
@@ -706,12 +759,12 @@ def run_eval_sampling(
         if graph["water"].num_nodes == 0:
             continue
 
-        out = flow_matcher.rk4_integrate(
+        # rk4 returns a per-frame trajectory (used for GIFs); euler does not.
```
Comment (DorisMai) [inline scripts/train.py]: why the distinction? maybe add a warning if user pass args to save gif

File: scripts/train.py [hunk 2]
```diff
@@ -980,6 +1041,8 @@ def save_checkpoint(
     optimizer_step_count,
     path,
     best=False,
+    best_val_loss=None,
+    best_sel_score=None,
```
Comment (DorisMai) [inline scripts/train.py:1050]: should add argument docstring. also, should you also store the selection metric? otherwise if upon reload metric is different, comparison of best becomes wrong, and should either raise or reset to some default value

File: scripts/train.py [hunk 3]
```diff
@@ -692,12 +737,20 @@ def run_eval_sampling(
         run_dir: Path to run directory for saving outputs
     """
     flow_matcher.model.eval()
+    # Fix the eval-sampling RNG so the per-epoch generative metrics are comparable
+    # across epochs (the water prior and integration both draw noise).
+    torch.manual_seed(args.val_seed)
```
Comment (DorisMai) [inline scripts/train.py]: If you run eval once, is this seed globally set such that any training step after this is affected? seems potentially problematic, and it would also ignore your `--seed -1` CLI arg.

File: scripts/train.py [hunk 4]
```diff
@@ -1126,6 +1192,14 @@ def main():
     """Run the full training pipeline."""
     args = parse_args()
 
+    # Seed weight init and data shuffling when requested; left unseeded otherwise.
+    # Seeded by default (--seed 42) for reproducibility; a negative seed opts out.
+    if args.seed >= 0:
+        random.seed(args.seed)
+        np.random.seed(args.seed)
```
Comment (DorisMai) [inline scripts/train.py:1215]: but this is overwritten on line 1293 to val_seed?

File: scripts/train.py [hunk 5]
```diff
@@ -1412,6 +1507,59 @@ def main():
...
+        elif eval_metrics:  # generative metric, defined only on eval epochs
+            if selection_metric == "blend":
+                raw = (
+                    0.85 * eval_metrics["eval/avg_f1"]
+                    + 0.15 * eval_metrics["eval/avg_auc_pr"]
+                )
+            else:
+                raw = eval_metrics[f"eval/avg_{selection_metric}"]
+            sel_history.append(raw)
+            sel = sum(sel_history[-3:]) / len(sel_history[-3:])  # rolling-3 smoothed
```
Comment (DorisMai) [inline scripts/train.py]: consider extract the hardcoded numbers as constants like BLEND_F1_WEIGHT and SEL_ROLLING_WINDOW, and update the helper str in argparse correspondingly. If you have justification for why these magic numbers that's even better.

**Review by DorisMai — APPROVED** [review]
mostly minor comments, a bit confusion on the seeds

File: scripts/train.py (hunk 5 above)
Comment (vratins) [reply scripts/train.py]: Will address in a docstring. The 0.85 is really just an arbitrary hp.

File: scripts/train.py (hunk 4 above)
Comment (vratins) [reply scripts/train.py:1215]: Fixed.

File: scripts/train.py (hunk 3 above)
Comment (vratins) [reply scripts/train.py]: Yes, fixed this.

File: scripts/train.py (hunk 2 above)
Comment (vratins) [reply scripts/train.py:1050]: Both done.

File: scripts/train.py (hunk 1 above)
Comment (vratins) [reply scripts/train.py]: Honestly just a development thing, I stopped using gifs to view flow when I moved to euler, will adjust this to return gif in either case for consistency.
